Automattic/wp-calypso
# Activity Log: Extract ActivityCard media previewing into its own component

Part of a continuing effort to modularize the ActivityCard component so it's easier to maintain.

#### Changes proposed in this Pull Request

* Create a new component `MediaPreview` to handle media previewing on Activity cards.
* Replace `renderMediaPreview()` in the `ActivityCard` component with this new component.
* Remove the `renderMediaPreview()` function since it's no longer used.
* Move media preview-specific styles into a `style.scss` file in the same directory as the component.

#### Testing instructions

**Prerequisites:** Create or gain access to a site with uploaded media -- at least one upload of a single file and one upload of multiple files -- in its Activity Log and/or Backup pages.

* In any Calypso experience, select your site and navigate to the Activity Log (and/or the Backup page, if the site has Backup Real-time).
* Verify that for activity items with media attached, the media preview(s) appear and function exactly as they do in production. Behavior and appearance should be identical. For events with multiple attached media files, you should see thumbnails for up to 3 items (or 5, if your screen is wide enough to be considered a "desktop").

#### Reference screenshots

##### Media preview without streams

<img height="300" src="https://user-images.githubusercontent.com/670067/126209970-9bbc9308-5873-4251-9b98-3a1b7b61c080.png"> <img height="300" src="https://user-images.githubusercontent.com/670067/126209966-e6df061d-dfc6-485e-a0e2-60a43d69f90d.png">

##### Media preview with streams

<img height="300" src="https://user-images.githubusercontent.com/670067/126209538-737743ec-b0f8-4d1d-8b9e-78a96cc688a9.png"> <img height="300" src="https://user-images.githubusercontent.com/670067/126209541-4f44e210-ad4e-497c-a596-99c7819416c0.png">
